app/src/main/java/com/asfoundation/wallet/ui/iab/BillingWebViewFragment.kt

```diff
@@ -0,0 +1,179 @@
+        return super.shouldOverrideUrlLoading(view, request)
+      }
+
+      override fun onPageFinished(view: WebView, url: String) {
+        super.onPageFinished(view, url)
+        if (!url.contains("/redirect")) {
+          val timeout = timeoutReference.getAndSet(null)
+          timeout?.cancel(false)
+          webview_progress_bar?.visibility = View.GONE
+          webview.visibility = View.VISIBLE
+        }
+        if (url.contains(ASYNC_PAYMENT_FORM_SHOWN_SCHEMA)) {
+          asyncDetailsShown = true
+        }
+      }
+    }
+    view.webview.settings.javaScriptEnabled = true
+    view.webview.loadUrl(currentUrl)
+    return view
+  }
+
+  fun handleBackPressed(): Boolean {
+    return if (asyncDetailsShown) {
+      webview.loadUrl("https://pay.dlocal.com/payment_method_connectors/global_pm//back")
+      asyncDetailsShown = false
+      true
+    } else {
+      false
+    }
+  }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    outState.putString(CURRENT_URL, currentUrl)
+  }
+
+  override fun onDestroy() {
+    executorService!!.shutdown()
+    super.onDestroy()
+  }
+
+  override fun onDetach() {
+    webViewActivity = null
+    webview?.webViewClient = null
+    super.onDetach()
+  }
+
+  private fun launchActivity(intent: Intent) {
+    try {
+      startActivity(intent)
+    } catch (exception: ActivityNotFoundException) {
+      exception.printStackTrace()
+      if (view != null) {
+        Snackbar.make(view!!, R.string.unknown_error,
+            Snackbar.LENGTH_SHORT)
+            .show()
+      }
+    }
+  }
+
+  private fun finishWithSuccess(url: String) {
+    val intent = Intent()
+    intent.data = Uri.parse(url)
+    webViewActivity!!.setResult(WebViewActivity.SUCCESS, intent)
+    webViewActivity!!.finish()
+  }
+
+  companion object {
+    private const val ADYEN_PAYMENT_SCHEMA = "adyencheckout://"
+    private const val LOCAL_PAYMENTS_SCHEMA = "myappcoins.com/t/"
+    private const val LOCAL_PAYMENTS_URL = "https://myappcoins.com/t/"
+    private const val GO_PAY_APP_PAYMENTS_SCHEMA = "gojek://gopay/merchanttransfer"
+    private const val LINE_APP_PAYMENTS_SCHEMA = "android-app://jp.naver.line.android"
+    private const val ASYNC_PAYMENT_FORM_SHOWN_SCHEMA = "https://pm.dlocal.com//v1/gateway/show?"
+    private const val CODAPAY_FINAL_REDIRECT_SCHEMA =
+        "https://airtime.codapayments.com/epcgw/dlocal/"
+    private const val CODAPAY_BACK_URL =
+        "https://pay.dlocal.com/payment_method_connectors/global_pm//back"
+    private const val URL = "url"
+    private const val CURRENT_URL = "currentUrl"
+    private const val ORDER_ID_PARAMETER = "OrderId"
+    fun newInstance(url: String?): BillingWebViewFragment {
+      val args = Bundle()
+      args.putString(URL, url)
+      val fragment = BillingWebViewFragment()
+      fragment.arguments = args
+      fragment.retainInstance = true
+      return fragment
```

Comment:
use apply and inline return

return BillingWebViewFragment().apply {
        arguments = args
        retainInstance = true
      }